docs/arch/adr-005-profiles-directories.md

```diff
@@ -0,0 +1,175 @@
+# Search order for `profiles.yml`
+
+## Context
+dbt has a solid foundation of [convention over configuration](https://en.wikipedia.org/wiki/Convention_over_configuration) (CoC), and this proposal would lean into this further.
+
+### Current behavior
+dbt needs a `profiles.yml` configuration file for database connection info. I believe the current search order is:
+1. `--profiles-dir` option
+1. `DBT_PROFILES_DIR` environment variable
+1. `~/.dbt/` directory
+
+dbt also needs a `dbt_project.yml`. The current order of precedence is:
+1. `--project-dir` option
+1. current working directory
+
+### Desired behavior
+Search order for `profiles.yml`:
+1. `--profiles-dir` option
+1. `DBT_PROFILES_DIR` environment variable
+1. current working directory (**NEW**)
+1. `~/.dbt/` directory
+
+Search order for `dbt_project.yml`:
+1. `--project-dir` option
+1. `DBT_PROJECT_DIR` environment variable (**NEW**)
+1. current working directory
+
+### General design requirements
+There's two necessary pieces for dbt to use a profile to connect to a target database:
+- property definitions for the desired target database
+- secrets to plug into the definition slots
+
+There are two main design requirements in terms of discoverability and accessibility:
+1. property definitions are easily discoverable (since they are not sensitive)
+1. secrets are non-discoverable and access is restricted per security policies
+
+#### Approach 1
```

Comment:
```suggestion
#### Approach 1 (current approach)
```

Reply:
I intended to describe that both approach 1 and 2 are possible currently, and each are probably used frequently.

1 would be more likely to occur in a local environment.
2 is probably used most often in continuous integration (CI) environments.

I will re-work this section for clarity.

What I probably should have used as a guiding light instead:
- Categorizing as **Approach A** or **Approach B** entirely depends on if _any_ secrets are in the `profiles.yml` file or not.

Whether there are any secrets depends whether it can be stored in VCS or not. Presence in VCS (or not) will then affect where the file is located within the file system.